db/system_keyspace.cc

```diff
@@ -3228,6 +3228,56 @@ future<mutation> system_keyspace::get_group0_history(distributed<service::storag
     co_return mutation(s, partition_key::from_singular(*s, GROUP0_HISTORY_KEY));
 }
 
+static constexpr auto GROUP0_UPGRADE_STATE_KEY = "group0_upgrade_state";
+
+future<service::group0_upgrade_state> system_keyspace::load_group0_upgrade_state() {
+    auto s = co_await get_scylla_local_param_as<sstring>(GROUP0_UPGRADE_STATE_KEY);
+
+    if (!s || *s == "use_old_procedures") {
```

Comment:
I am against using "old" and "new" in variable names. Over time everything gets old or very old. So if we ever add a new procedure, we'll have to give it some very stupid name, like "new new" or "real" or "best". Better use more concrete definitions, like "use_gossip_for_schema" or "use_raft_for_schema", "4.x" vs "5.x" and so on.

Reply:
`use_pre_raft_procedures`, `use_post_raft_procedures`?